src/routes/plans/[id]/+page.svelte

```diff
@@ -104,7 +107,26 @@
     $view = { ...data.initialView };
   }
 
-  $: $activitiesMap = createActivitiesMap($plan, $activityDirectives, $simulationSpans);
+  $: if ($plan) {
+    effects
+      .getResourcesExternal($plan.id, $plan.start_time, $plan.duration)
+      .then(newResources => ($externalResources = newResources));
+  }
+
+  $: if ($plan && $simulationDataset !== undefined) {
```

Comment:
Am I understanding correctly that this fires whenever either `$plan` or `$simulationDataset` is updated? So - when you click the "simulate" button - this will fire once on sim start, and then once again on sim finish. It will also fire when you use the dropdown to select a different dataset.

If so, that sounds great to me :+1:

Reply:
Yup precisely!